scripts/repo-variable-writer.go

```diff
@@ -0,0 +1,373 @@
+			Version        string `json:"version"`
+			CncfProjectSub string `json:"cncf_project_sub"`
+		} `json:"inputs"`
+	}
+
+	for _, subComponent := range projData.SubComponents {
+		workflowData := WorkflowData{}
+		workflowData.Inputs.CncfProject = projData.Name
+		workflowData.Ref = "main" // TODO: need to figure it out
+		workflowData.Inputs.Version = version
+		workflowData.Inputs.CncfProjectSub = subComponent
+
+		var _newWorkflowData bytes.Buffer
+
+		if err := json.NewEncoder(&_newWorkflowData).Encode(workflowData); err != nil {
+			return fmt.Errorf("failed to serialize the body: %v", err)
+		}
+
+		resp, err := handleHTTPCall(
+			http.MethodPost,
+			*url,
+			time.Minute,
+			&_newWorkflowData,
+			header,
+		)
+		if err != nil {
+			return err
+		}
+
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusNoContent {
+			return fmt.Errorf(
+				"failed in workflowDispatch, status code was not 204, got: %v",
+				resp.StatusCode)
+		}
+
+		slog.Info("Successfully workflow_dispatch",
+			"Proj", projData.Name,
+			"Org", projData.Organization,
+			"Ver", version,
+			"Subcomponent", subComponent)
+	}
+
+	return nil
+}
+
+func main() {
+
+	projects, err := loadProjectMetadata(LocProjMeta)
+	if err != nil {
+		slog.Error("Failed: unable to read the projects Metadata", "Reason", err)
+		os.Exit(1)
+	}
+	slog.Info("Got projects metadata", "Projects", projects)
+
+	storage, err := NewGithubVariable()
+	if err != nil {
+		slog.Error("Failed: initialize storage", "Reason", err)
+		os.Exit(1)
+	}
+
+	wg := new(sync.WaitGroup)
+	wg.Add(len(projects.Projects))
+
+	errChan := make(chan error, len(projects.Projects))
+
+	for _, project := range projects.Projects {
+
+		go func(proj Project) {
+			defer wg.Done()
+
+			latestRelease, err := fetchLatestRelease(
+				proj.Organization,
+				proj.Name,
+			)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			storedRelease, err := storage.ReadRepoVariable(proj.Name)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			if strings.Compare(*storedRelease, *latestRelease) != 0 {
+				slog.Info("Release Versions", "Proj", proj.Name, "Org", proj.Organization, "Latest", *latestRelease, "Current", *storedRelease)
+				if err := storage.UpdateRepoVariable(proj.Name, *latestRelease); err != nil {
+					errChan <- err
+					return
+				}
+				slog.Info("Updated to Latest version", "Proj", proj.Name, "Org", proj.Organization, "Ver", *latestRelease)
+
+				if err := storage.UpdatedProjectDispatcher(
+					proj,
+					*latestRelease,
+				); err != nil {
+					errChan <- err
+					return
+				}
+
+			} else {
+				slog.Info("Already in Latest version", "Proj", proj.Name, "Org", proj.Organization, "Ver", *latestRelease)
+			}
+		}(project)
+	}
+
+	wg.Wait()
+	close(errChan)
+
+	hadFailures := false
+	for err := range errChan {
+		if err != nil {
+			hadFailures = true
+			slog.Error("Failed", "Reason", err)
+		}
+	}
+	if hadFailures {
+		os.Exit(1)
+	}
+
+	slog.Info("Finished execution")
```

Comment:
this does not provide any information. we can either remove this log or add some kind of info to it